### PR TITLE
Improve test output, fix docstring

### LINKED ISSUE #562: Inaccurate docstring for `TestChainFragmentFork`
The docstring for `TestChainFragmentFork` says

```
-- For example: start with a whole chain, then take a fragment of that chain.
-- Take a fragment of that fragment to obtain a second chain fragment. The two
-- fragments will share identical blocks, but one fragment may contain more
-- than the other. Now add some unique blocks to each fragment to obtain two
-- forks. These are the four fragments generated 'TestChainFragmentFork'.
```

but an example generated value is

```haskell
      TestChainFragmentFork
          ChainFragment:
      
          ChainFragment:
              Block {
    blockHeader = BlockHeader {
      headerHash = HeaderHash 1328581587779029793
    , headerPrevHash = GenesisHash
    , headerSlot = SlotNo {unSlotNo = 1}
    , headerBlockNo = BlockNo {unBlockNo = 1}
    , headerSigner = BlockSigner 1
    , headerBodyHash = BodyHash (-608257208647770408)
  }
  , blockBody = BlockBody "EMPTY"
}
      
          ChainFragment:
              Block {
    blockHeader = BlockHeader {
      headerHash = HeaderHash 1328581587779029793
    , headerPrevHash = GenesisHash
    , headerSlot = SlotNo {unSlotNo = 1}
    , headerBlockNo = BlockNo {unBlockNo = 1}
    , headerSigner = BlockSigner 1
    , headerBodyHash = BodyHash (-608257208647770408)
  }
  , blockBody = BlockBody "EMPTY"
}
      
          ChainFragment:
              Block {
    blockHeader = BlockHeader {
      headerHash = HeaderHash 1328581587779029793
    , headerPrevHash = GenesisHash
    , headerSlot = SlotNo {unSlotNo = 1}
    , headerBlockNo = BlockNo {unBlockNo = 1}
    , headerSigner = BlockSigner 1
    , headerBodyHash = BodyHash (-608257208647770408)
  }
  , blockBody = BlockBody "EMPTY"
}
```

This violates the docstring in two ways:

* The second fragment is not a prefix of the first
* The blocks that got added to these fragments are not unique.

### PR DESCRIPTION
**Pull request by edsko**
This adds the GHC callstack to a few functions for better error reporting, and improves the feedback we get from some of the network tests.

Also fixes the description of one of the tests (closes #562).

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/ChainFragment.hs [hunk 1]
```diff
@@ -201,25 +205,45 @@ valid (c :> b) = valid c && validExtension c b
...
+isValidSuccessorOf bSucc b = isRight $ isValidSuccessorOf' bSucc b
+
+-- | Variation on 'isValidSuccessorOf' that provides more information
+isValidSuccessorOf' :: (HasCallStack, HasHeader block)
+                    => block  -- ^ @bSucc@
+                    -> block  -- ^ @b@
+                    -> Either String ()
+isValidSuccessorOf' bSucc b
+  | pointHash p /= blockPrevHash bSucc
+  = Left $ "prevHash (" <> show (blockPrevHash bSucc) <> ") doesn't match hash of tip (" <> show (pointHash p) <> ")" ++ " at " ++ prettyCallStack callStack
```
Comment (dcoutts) [inline ouroboros-network/src/Ouroboros/Network/ChainFragment.hs]: Any chance we can stick to 80 cols?

Comment (edsko) [reply ouroboros-network/src/Ouroboros/Network/ChainFragment.hs]: Ok.

File: ouroboros-network/test/Test/ChainFragment.hs [hunk 2]
```diff
@@ -225,16 +225,17 @@ prop_lookupBySlot (TestChainFragmentAndPoint c p) =
...
 prop_intersectChainFragments (TestChainFragmentFork origL1 origL2 c1 c2) =
-  case CF.intersectChainFragments c1 c2 of
+  let intersection = CF.intersectChainFragments c1 c2 in
+  counterexample ("intersection: " ++ show intersection) $
+  case intersection of
     Nothing -> (CF.null origL1 || CF.null origL2)
-            && L.intersect (CF.toNewestFirst c1) (CF.toNewestFirst c2) == []
+            .&&. L.intersect (CF.toNewestFirst c1) (CF.toNewestFirst c2) === []
     Just (l1, l2, r1, r2) ->
-      l1 == origL1 && l2 == origL2 &&
```
Comment (mrBliss) [inline ouroboros-network/test/Test/ChainFragment.hs]: Here is a sneaky fix of the test itself :slightly_smiling_face:

Comment (edsko) [reply ouroboros-network/test/Test/ChainFragment.hs]: Sneaky? 我? :angel:

Comment (edsko) [reply ouroboros-network/test/Test/ChainFragment.hs]: I suppose I neglected to mention that in the PR description. Oops :flushed:

**Pull request merged**